Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2727 @@
+     */
+    public var theme: Theme = .black {
+        didSet {
+            //
+            // Set the color based on
+            // selected option
+            var tintColor: UIColor
+            switch theme {
+            case .black:
+                tintColor = UIColor.black
+                backgroundColor = UIColor.lightGray.withAlphaComponent(0.6)
+            case .blue:
+                tintColor = UIColor.customBlue
+                backgroundColor = UIColor.lightSkyBlue.withAlphaComponent(0.3)
+            case .oceanBlue:
+                tintColor = UIColor.oceanBlue
+                backgroundColor = UIColor.white.withAlphaComponent(0.6)
+            }
+            
+            // Set colors of the components
+            currentExtentFillColor = tintColor
+            currentExtentLabelColor = tintColor
+            fullExtentBorderColor = tintColor
+            fullExtentFillColor = UIColor.white
+            fullExtentLabelColor = tintColor
+            timeStepIntervalLabelColor = tintColor
+            timeStepIntervalTickColor = tintColor
+            thumbFillColor = UIColor.white
+            thumbBorderColor = tintColor
+            layerExtentFillColor = tintColor.withAlphaComponent(0.4)
+            pinnedThumbFillColor = tintColor
+            playbackButtonsFillColor = tintColor
+        }
+    }
+    
+    /**
+     A Boolean value indicating whether changes in the slider’s value
+     generate continuous update events. Default is True.
+     */
+    public var isContinuous: Bool = true
+    
+    // MARK: - Private Properties
+    
+    private var currentExtentStartTime: Date?
+    private var currentExtentEndTime: Date?
+    private var previousCurrentExtentStartTime: Date?
+    private var previousCurrentExtentEndTime: Date?
+    private var startTimeStepIndex: Int = -1
+    private var endTimeStepIndex: Int = -1
+    private var internalUpdate: Bool = false
+    
+    fileprivate var layerExtent: AGSTimeExtent?
+    
+    private let trackLayer = TimeSliderTrackLayer()
+    private let tickMarkLayer = TimeSliderTickMarkLayer()
+    private let lowerThumbLayer = TimeSliderThumbLayer()
+    private let upperThumbLayer = TimeSliderThumbLayer()
+    private let fullExtentStartTimeLabel: CATextLayer = CATextLayer()
+    private let fullExtentEndTimeLabel: CATextLayer = CATextLayer()
+    private let currentExtentStartTimeLabel: CATextLayer = CATextLayer()
+    private let currentExtentEndTimeLabel: CATextLayer = CATextLayer()
+    
+    private let minimumFrameWidth: CGFloat = 250.0
+    private let maximumThumbSize: CGFloat = 50.0
+    private var trackLayerSidePadding: CGFloat = 30.0
+    private let labelSidePadding: CGFloat = 10.0
+    private let labelPadding: CGFloat = 3.0
+    private let paddingBetweenLabels: CGFloat = 5.0
+    
+    private var tickMarkLabels = [CATextLayer]()
+    fileprivate var tickMarks = [TickMark]()
+    
+    private var playPauseButton = UIButton(type: .custom)
```

Comment:
These button properties are never mutated after initialization. Could you make them constants (`let`)?

Reply:
fixed